### PR TITLE
add ghost v2 docker images

### LINKED ISSUE #142: Ghost 2.0 release
Hi everyone,

Ghost 2.0 has been released now.

Will this be added to the docker-library as well, possibly with an upgrade path?

Cheers,
Mark

### PR DESCRIPTION
**Pull request by acburdine**
closes #142 

Main difference from ghost v1 is that knex-migrator is no longer needed outside of Ghost - Ghost itself will run it on boot.

### CODE REVIEW

File: 2/alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# make a config.json symlink for NODE_ENV=development (and sanity check that it's correct)
+	su-exec node ln -s config.production.json "$GHOST_INSTALL/config.development.json"; \
+	readlink -f "$GHOST_INSTALL/config.development.json"; \
+	\
+# need to save initial content for pre-seeding empty volumes
+	mv "$GHOST_CONTENT" "$GHOST_INSTALL/content.orig"; \
+	mkdir -p "$GHOST_CONTENT"; \
+	chown node:node "$GHOST_CONTENT"
+
+# TODO multiarch sqlite3 (once either "node:6-alpine" has multiarch or we switch to a base that does)
```
Comment (tianon) [inline 2/alpine/Dockerfile]: Heh, probably time to revisit this (although not going to hold this up to do so). :+1:

**Comment by tianon** [comment]
Here's a slightly better diff: (since GitHub doesn't support `--find-copies-harder` through the web interface)

```diff
diff --git a/.travis.yml b/.travis.yml
index ac01164..31869e1 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -2,6 +2,8 @@ language: bash
 services: docker
 
 env:
+  - VERSION=2 VARIANT=debian
+  - VERSION=2 VARIANT=alpine
   - VERSION=1 VARIANT=debian
   - VERSION=1 VARIANT=alpine
   - VERSION=0 VARIANT=debian
diff --git a/1/alpine/Dockerfile b/2/alpine/Dockerfile
similarity index 77%
copy from 1/alpine/Dockerfile
copy to 2/alpine/Dockerfile
index 847add9..1fc8a6c 100644
--- a/1/alpine/Dockerfile
+++ b/2/alpine/Dockerfile
@@ -17,7 +17,7 @@ RUN npm install -g "ghost-cli@$GHOST_CLI_VERSION"
 ENV GHOST_INSTALL /var/lib/ghost
 ENV GHOST_CONTENT /var/lib/ghost/content
 
-ENV GHOST_VERSION 1.25.5
+ENV GHOST_VERSION 2.0.3
 
 RUN set -ex; \
 	mkdir -p "$GHOST_INSTALL"; \
@@ -37,14 +37,7 @@ RUN set -ex; \
 # need to save initial content for pre-seeding empty volumes
 	mv "$GHOST_CONTENT" "$GHOST_INSTALL/content.orig"; \
 	mkdir -p "$GHOST_CONTENT"; \
-	chown node:node "$GHOST_CONTENT"; \
-	\
-# sanity check to ensure knex-migrator was installed
-	"$GHOST_INSTALL/current/node_modules/knex-migrator/bin/knex-migrator" --version
-
-# add knex-migrator bins into PATH
-# we want these from the context of Ghost's "node_modules" directory (instead of doing "npm install -g knex-migrator") so they can share the DB driver modules
-ENV PATH $PATH:$GHOST_INSTALL/current/node_modules/knex-migrator/bin
+	chown node:node "$GHOST_CONTENT"
 
 # TODO multiarch sqlite3 (once either "node:6-alpine" has multiarch or we switch to a base that does)
 
diff --git a/1/alpine/docker-entrypoint.sh b/2/alpine/docker-entrypoint.sh
similarity index 89%
copy from 1/alpine/docker-entrypoint.sh
copy to 2/alpine/docker-entrypoint.sh
index 21eee25..bedd4fa 100755
--- a/1/alpine/docker-entrypoint.sh
+++ b/2/alpine/docker-entrypoint.sh
@@ -17,8 +17,6 @@ if [[ "$*" == node*current/index.js* ]]; then
 			tar -cC "$(dirname "$src")" "$(basename "$src")" | tar -xC "$(dirname "$target")"
 		fi
 	done
-
-	knex-migrator-migrate --init --mgpath "$GHOST_INSTALL/current"
 fi
 
 exec "$@"
diff --git a/1/debian/Dockerfile b/2/debian/Dockerfile
similarity index 87%
copy from 1/debian/Dockerfile
copy to 2/debian/Dockerfile
index a834190..8c8e63a 100644
--- a/1/debian/Dockerfile
+++ b/2/debian/Dockerfile
@@ -23,7 +23,7 @@ RUN npm install -g "ghost-cli@$GHOST_CLI_VERSION"
 ENV GHOST_INSTALL /var/lib/ghost
 ENV GHOST_CONTENT /var/lib/ghost/content
 
-ENV GHOST_VERSION 1.25.5
+ENV GHOST_VERSION 2.0.3
 
 RUN set -ex; \
 	mkdir -p "$GHOST_INSTALL"; \
@@ -43,14 +43,7 @@ RUN set -ex; \
 # need to save initial content for pre-seeding empty volumes
 	mv "$GHOST_CONTENT" "$GHOST_INSTALL/content.orig"; \
 	mkdir -p "$GHOST_CONTENT"; \
-	chown node:node "$GHOST_CONTENT"; \
-	\
-# sanity check to ensure knex-migrator was installed
-	"$GHOST_INSTALL/current/node_modules/knex-migrator/bin/knex-migrator" --version
-
-# add knex-migrator bins into PATH
-# we want these from the context of Ghost's "node_modules" directory (instead of doing "npm install -g knex-migrator") so they can share the DB driver modules
-ENV PATH $PATH:$GHOST_INSTALL/current/node_modules/knex-migrator/bin
+	chown node:node "$GHOST_CONTENT"
 
 RUN set -eux; \
 # force install "sqlite3" manually since it's an optional dependency of "ghost"
diff --git a/1/debian/docker-entrypoint.sh b/2/debian/docker-entrypoint.sh
similarity index 89%
copy from 1/debian/docker-entrypoint.sh
copy to 2/debian/docker-entrypoint.sh
index 1060b05..c7bfbe3 100755
--- a/1/debian/docker-entrypoint.sh
+++ b/2/debian/docker-entrypoint.sh
@@ -17,8 +17,6 @@ if [[ "$*" == node*current/index.js* ]]; then
 			tar -cC "$(dirname "$src")" "$(basename "$src")" | tar -xC "$(dirname "$target")"
 		fi
 	done
-
-	knex-migrator-migrate --init --mgpath "$GHOST_INSTALL/current"
 fi
 
 exec "$@"
```

**Comment by tianon** [comment]
(The parts that are gone are some of the parts that have been the most annoying!! :tada: :heart:)

**Comment by crespum** [comment]
The image works when deploying a new Ghost instance, however it doesn't work when updating from a previous Ghost v1 blog because it doesn't migrate the old DB. To be more precise, the problem is the new `column_id` field. Is there any way to solve this?

**Comment by BirkhoffLee** [comment]
Same as @crespum here. Ghost's seemly not migrating anything.

**Comment by tianon** [comment]
See https://github.com/docker-library/ghost/issues/147, especially https://github.com/docker-library/ghost/issues/147#issuecomment-417372462:

> Did you by chance try to update to 2.x from a 1.x release < 1.25.4? In Ghost-CLI we added a provision that prevented updates if you were not already running the latest 1.x version of Ghost, and my guess is that the errors you're seeing are related to the fact that you need to upgrade to the latest 1.x version first before trying the 2.x upgrade.

**Pull request merged**